apps/web/lib/reschedule/[uid]/getServerSideProps.ts

```diff
@@ -115,6 +115,28 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
     };
   }
 
+  const eventType = booking.eventType ? booking.eventType : getDefaultEvent(dynamicEventSlugRef);
+
+  const enrichedBookingUser = booking.user
+    ? await UserRepository.enrichUserWithItsProfile({ user: booking.user })
+    : null;
+
+  const eventUrl = await buildEventUrlFromBooking({
+    eventType,
+    dynamicGroupSlugRef: booking.dynamicGroupSlugRef ?? null,
+    profileEnrichedBookingUser: enrichedBookingUser,
+  });
+
+  const isBookingInPast = new Date(booking.endTime) < new Date();
```

Comment:
What if `booking.endTime` is `null`, `undefined`, or simply invalid? Maybe it's best to add a small additional check like this:

```
const isBookingInPast = booking.endTime && new Date(booking.endTime) < new Date();
if (isBookingInPast) {
  return {
    redirect: {
      destination: eventUrl,
      permanent: false,
    },
  };
}
```

Reply:
yes, it's good to have null check